test/refactoredTestSetup.ts

```diff
@@ -0,0 +1,336 @@
+    } catch (error) {
+      utilities.log(
+        `Deleting scratch org (or info) failed with Error: ${(error as Error).message}`
+      );
+    }
+  }
+
+  public async setUpTestingWorkspace(projectConfig: ProjectConfig) {
+    utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingWorkspace()...`);
+    let projectName;
+    switch (projectConfig.projectShape) {
+      case ProjectShapeOption.NEW:
+        if (!fs.existsSync(this.tempFolderPath)) {
+          utilities.createFolder(this.tempFolderPath);
+        }
+        await utilities.generateSfProject(this.tempProjectName, this.tempFolderPath); // generate a sf project for 'new'
+        this.projectFolderPath = path.join(this.tempFolderPath, this.tempProjectName);
+        break;
+
+      case ProjectShapeOption.NAMED:
+        if (projectConfig.githubRepoUrl) {
+          // verify if folder matches the github repo url
+          const repoExists = await utilities.gitRepoExists(projectConfig.githubRepoUrl);
+          if (!repoExists) {
+            this.throwError(`Repository does not exist or is inaccessible: ${projectConfig.githubRepoUrl}`);
+          }
+          const repoName = utilities.getRepoNameFromUrl(projectConfig.githubRepoUrl);
+          projectName = repoName;
+          if (projectConfig.folderPath) {
+            const localProjName = utilities.getFolderName(projectConfig.folderPath);
+            if (localProjName != repoName) {
+              this.throwError(`the local project ${projectConfig.githubRepoUrl} does not match the required Github repo ${projectConfig.githubRepoUrl}`);
+            } else {
+              // if it is a match, use the local folder directly. Local dev use only.
+              this.projectFolderPath = projectConfig.folderPath;
+            }
+          } else {
+            // clone the project from Github URL directly
+            this.projectFolderPath = path.join(this.tempFolderPath, repoName!);
+            await utilities.gitClone(projectConfig.githubRepoUrl, this.projectFolderPath);
+          }
+        } else {
+          // missing info, throw an error
+          this.throwError(`githubRepoUrl is required for named project shape`);
+        }
+        break;
+
+      case ProjectShapeOption.ANY:
+        // ANY: workspace is designated to open when wdio is initialized
+        if (projectConfig.folderPath) {
+          this.projectFolderPath = projectConfig.folderPath;
+          projectName = utilities.getFolderName(projectConfig.folderPath);
+        } else this.throwError('folder path is required for named project!');
+        return;
+
+      case ProjectShapeOption.NONE:
+        // NONE: blank project by default
+        /* create the e2e-temp folder to benefit further testing */
+        if (!fs.existsSync(this.tempFolderPath)) {
+          utilities.createFolder(this.tempFolderPath);
+        }
+        this.projectFolderPath = path.join(this.tempFolderPath, this.tempProjectName);
```

Comment:
What is the value of `this.tempProjectName` at this moment? Seems like it would be empty string. Is this what we want. Also, if the test never creates a project, where will screnshots of failures be saved and associated with the current test.

Reply:
You are right the projectFolderPath here should be undefined. 

The screenshots should be saved according to the test title and the test step name. So it has nothing to do with if a project is open or not.